Enhanced documentation on batch datasets (#514)
PowerGridModel/power-grid-model: docs/user_manual/dataset-terminology.md

## Data structures

- **Dataset:** Either a single or a batch dataset.
- **SingleDataset:** A data type storing input data (ie. all elements of all components) for a single scenario.
- **BatchDataset:** A data type storing update and or output data for one or more scenarios. A batch dataset can be either a sparse or a dense one.

>     - **SparseBatchDataset:**  Dictionaries with a one-dimensional numpy int32 array and a one-dimensional structured numpy. 

Reviewer: ```suggestion
    - **SparseBatchDataset:**  Dictionaries with a one-dimensional numpy int32 array and a one-dimensional structured numpy arrays.
```

> ## Data structures

Reviewer: Since you now have the API references: can you reference/link them please?

- **Dataset:** Either a single or a batch dataset.

>     - **SingleDataset:** A data type storing input data (ie. all elements of all components) for a single scenario.

Reviewer: ```suggestion
    - **SingleDataset:** A data type storing input data (i.e. all elements of all components) for a single scenario.
```

>     - **BatchDataset:** A data type storing update and or output data for one or more scenarios. A batch dataset can be a sparse or dense one.

Reviewer 2: ```suggestion
    - **BatchDataset:** A data type storing update and or output data for one or more scenarios. Batch datasets can be sparse or dense.
```
or
```suggestion
    - **BatchDataset:** A data type storing update and or output data for one or more scenarios. A batch dataset can be either a sparse one or a dense one.
```

>         - **SparseBatchDataset:**  Dictionaries with a one-dimensional numpy int32 array and a one-dimensional structured numpy. 

Author: We don't highlight these words throughout the documentation.
